NebulousConquestHelper/src/bot/Bot.cs

```diff
@@ -0,0 +1,165 @@
+
+namespace NebulousConquestHelper
+{
+    public class Bot
+    {
+        public readonly EventId BotEventId = new EventId(42, "Nebulous-QB");
+
+        public DiscordClient Client { get; set; }
+        public CommandsNextExtension Commands { get; set; }
+
+
+		public async Task RunBotAsync()
+		{
+			String configFileName = "src/data/config.json";
+			String json = "";
+			if (!File.Exists(configFileName))
+			{
+				Console.WriteLine("config.json is missing. Copy config.example.json and put in the discord bot token");
+				return;
+			}
+			using (FileStream fs = File.OpenRead(configFileName))
+			using (StreamReader sr = new StreamReader(fs, new UTF8Encoding(false)))
+				json = await sr.ReadToEndAsync();
+
+			BotConfig cfgjson = JsonConvert.DeserializeObject<BotConfig>(json);
+			if (string.IsNullOrEmpty(cfgjson.Token))
+			{
+				Console.WriteLine("Missing token in config");
+				return;
+			}
+			DiscordConfiguration cfg = new DiscordConfiguration
+			{
+				Token = cfgjson.Token,
+				TokenType = TokenType.Bot,
+				Intents = DiscordIntents.AllUnprivileged 
+					| DiscordIntents.DirectMessages 
+					| DiscordIntents.DirectMessageTyping 
+					| DiscordIntents.GuildMessages 
+					| DiscordIntents.GuildMessageTyping,
+
+				AutoReconnect = true,
+				MinimumLogLevel = LogLevel.Debug,
+			};
+
+			Client = new DiscordClient(cfg);
+
+			Client.Ready += Client_Ready;
+			Client.GuildAvailable += Client_GuildAvailable;
+			Client.ClientErrored += Client_ClientErrored;
+            Client.InteractionCreated += Client_InteractionCreated;
+
+
+			var ccfg = new CommandsNextConfiguration
+			{
+				StringPrefixes = new[] { cfgjson.CommandPrefix },
+				EnableDms = true,
+				EnableMentionPrefix = true
+			};
+
+			Commands = Client.UseCommandsNext(ccfg);
+
+			Commands.CommandExecuted += Commands_CommandExecuted;
+			Commands.CommandErrored += Commands_CommandErrored;
+
+			Commands.RegisterCommands<BotCommands>();
+
+			Commands.SetHelpFormatter<BotHelpFormatter>();
+
+			await Client.ConnectAsync();
+
+			await Task.Delay(-1);
+		}
+
+        private async Task Client_InteractionCreated(DiscordClient sender, InteractionCreateEventArgs e)
+        {
+			Client.Logger.LogInformation(BotEventId, $"{e.Interaction.User.Username} sent an interaction: {e.Interaction.Data.Name}");
+
+            try
+            {
+				await e.Interaction.CreateResponseAsync(InteractionResponseType.DeferredChannelMessageWithSource);
+			}
+            catch (Exception ex)
+            {
+				Client.Logger.LogError(BotEventId, "Exception when responding to interaction");
+                throw;
+            }
+
+            try
+            {
+				using (FileStream fs = new FileStream(Helper.DATA_FOLDER_PATH + "SystemMap.png", FileMode.Open, FileAccess.Read))
+				{
+					await e.Interaction.CreateFollowupMessageAsync(new DiscordFollowupMessageBuilder()
+						.WithContent("Here is the map for the current game state:")
+						.AddFile("SystemMap.png", fs));
```

Comment:
Absolutely it should. I was waiting to have a discussion on how the 'systems' should be included. Since the mapper is static I can easily add that here.

Reply:
Requested change has been made.